### PR TITLE
Full Screen Editing Mode

### PR DESCRIPTION
**Pull request by RobertDenis**
Adding maximize/minimize functionality to the editor screen.
Adding dynamic year to sidenav copyright.

Please note that I used absolute positioning in full screen mode so that the editor could sit on top of the map.

### CODE REVIEW

**Comment by eshon** [comment]
@ischneider do you have time to review this PR as well?

**Comment by eshon** [comment]
Also I'm concerned about the CSS here.

**Comment by RobertDenis** [comment]
Jira #78, 182, 322, and 323.

**Comment by eshon** [comment]
There's to much stuff to review in this single PR.  In the future, please separate them.  I'll review this PR later this week then so don't change it yet.

File: geoserver/webapp/app/maps/detail/compose.js [hunk 1]
```diff
@@ -124,6 +144,117 @@ angular.module('gsApp.maps.compose', [
         });
       };
 
+      $scope.maximize = function() {
+        var screenWidth = $window.innerWidth;
+        var screenHeight = $window.innerHeight;
+
+        $scope.navbarHeight = angular.element('#navbar').height();
```
Comment (ischneider) [inline geoserver/webapp/app/maps/detail/compose.js]: don't put things in the `$scope` unless you need to. if you simply need to manage state (as it appears you are doing), use something in the controller (like declare an object like `var resizeState = {}`)

File: geoserver/webapp/app/maps/detail/compose.js [hunk 2]
```diff
@@ -124,6 +144,117 @@ angular.module('gsApp.maps.compose', [
...
+        $scope.resizerWidth = angular.element('#resizer').width();
+        var offsetTop = 50;
+        var offsetBottom = 150;
+        var offsetRight = 50;
+        var newEditingHeight = screenHeight - offsetTop - offsetBottom;
+        var editorWidth = 650;
+        $scope.showMaximize = false;
+
+        //Remove nav, sidenav, resizer and info bars
+        angular.element('#navbarWrapper').css({
```
Comment (ischneider) [inline geoserver/webapp/app/maps/detail/compose.js]: while you _can_ do dom manipulation in the controller, pay attention to the guidelines here: https://code.angularjs.org/1.2.28/docs/guide/controller (specifically the part that says 'do not')

Most, if not all, of these manipulations can be done directly in the template - this makes things less fragile (id typos, duplicate ids, etc.) and reduces your efforts (typing ids in two+ places). Look at `ng-hide`, `ng-show`, `ng-if`, and `ng-class`.

Pixel level manipulations can also be done via inline style attributes and template variables (e.g.`style='width:{{width}}px'` )

**Comment by tbarsballe** [comment]
This has been acomplished by #886 - closing.

**Pull request closed without merging**